app/src/main/java/com/example/simbirsoft_android_practice/core/RepositoryProvider.kt

```diff
@@ -0,0 +1,6 @@
+package com.example.simbirsoft_android_practice.core
+
+interface RepositoryProvider {
+    val newsRepository: NewsRepository
+    val categoryRepository: CategoryRepository
+}
```

Comment:
Не стесняйся делать удобные инструменты для личного пользования
```suggestion
interface RepositoryProvider {
    val newsRepository: NewsRepository
    val categoryRepository: CategoryRepository
} {
    companion object {
         fun fromContext(context: Context): RepositoryProvider 
             = (context.applicationContext) as? RepositoryProvider 
                 ?: error("RepositoryProvider was requested before application was ready")
         }
    }
}
```